boards: stm32f746g_disco: Enable a few peripherals

chrta opened the pull request:
This enables I2C, SPI, PWM, RTC and USB.

RTC tested on samples/drivers/rtc
USB tested on samples/subsys/usb/cdc_acm

Open issues:
1. I2C and SPI are not tested, due to the lack of hardware.

chrta committed:
boards: nucleo_f429zi: Fix typo in code comments

Fix minor typo in code comments of pinmux.c

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
drivers/rtc: Enable RTC driver for STM32F7 series

Enable RTC driver for STM32F7 series.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
drivers: spi_ll_stm32: Enable SPI driver for F7 family

Enables SPI driver for STM32F7 SoCs

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
boards: stm32f746g_disco: Enable a few peripherals

This enables I2C, SPI, PWM, RTC and USB.

RTC tested on samples/drivers/rtc
USB tested on samples/subsys/usb/cdc_acm

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta commented:
Pinmux defines for SPI are missing.

erwango commented on dts/arm/st/f7/stm32f7.dtsi:
```diff
@@ -241,6 +241,17 @@
 			label = "I2C_3";
 		};
 
+		spi2: spi@40003800 {
```
Please populate all SPI devices applicable STM32F746xg

chrta replied:
done

chrta committed:
drivers: pinmux: stm32f7: Add SPI2 pinmux entries

This patch adds the SPI2 pinmux entries for the STM32F7 series.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta committed:
dts/arm/st_stm32: Add SPI nodes to STM32 F7 series

This patch adds SPI nodes to STM32 F7 series, as well as the
corresponding dts fixup entries.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

chrta commented on soc/arm/st_stm32/stm32f7/dts_fixup.h:
```diff
@@ -134,4 +134,10 @@
+#define CONFIG_RTC_0_IRQ                        ST_STM32_RTC_40002800_IRQ_0
+#define CONFIG_RTC_0_NAME                       ST_STM32_RTC_40002800_LABEL
+#define CONFIG_RTC_PRESCALER                    ST_STM32_RTC_40002800_PRESCALER
+
```
done

erwango commented:
@chrta : I'll push a PR to fix stm32cube issue

chrta commented:
Rebased

erwango commented on boards/arm/stm32f746g_disco/pinmux.c:
```diff
@@ -35,6 +38,20 @@ static const struct pin_config pinconf[] = {
+#endif  /* CONFIG_I2C_1 */
+#ifdef CONFIG_SPI_2
+	{STM32_PIN_PI0,  STM32F7_PINMUX_FUNC_PI0_SPI2_NSS},
+	{STM32_PIN_PI1,  STM32F7_PINMUX_FUNC_PI1_SPI2_SCK},
```
Is STM32F7_PINMUX_FUNC_PI1_SPI2_SCK declared in drivers/pinmux/stm32/pinmux_stm32f7.h?

chrta replied:
Yes, i added all SPI2 pinmux definitions to this file.

galak commented:
@dbkinder can you review the docs.

dbkinder reviewed (approved):
+1 for doc changes, thanks.
